src/python/grpcio/grpc/aio/_base_call.py

```diff
@@ -86,15 +86,15 @@ class Call(RpcContext, metaclass=ABCMeta):
     """The abstract base class of an RPC on the client-side."""
 
     @abstractmethod
-    async def initial_metadata(self) -> Metadata:
+    async def initial_metadata(self) -> Optional[Metadata]:
```

Comment:
Maybe we can use `MetadataType` in `_typing.py`? And update the metadata type to support both kind of metadata.

```python
MetadataType = Union[Iterable[MetadatumType], Metadata]
```

Reply:
While I think that this would lead to greater flexibility, I would like to mention that since this is the `_base_call` of the AIO package, it (from what I understand) should always return `Metadata`. Perhaps it would be useful to define a shared `typing.Protocol` across the AIO and standard implementations for classes that are meant to be used concurrently.

Reply:
> While I think that this would lead to greater flexibility, I would like to mention that since this is the _base_call of the AIO package, it (from what I understand) should always return Metadata.

My bad. I mistaken this PR to be changing the input metadata typing. For metadata provided by the AIO library, it should be `Metadata`.

> typing.Protocol?

We want to support 3.6, so we can't use 3.8 features yet.

---

In that case, can you be more specific about why you want to add the `Optional` annotation here? In your description, you mentioned this gives interceptors freedom to return optional Metadata. Generally, the interceptors can also cope to the existing type annotation by returning an empty `Metadata` object.

If you think this type annotation is important, we need to do this correctly with some extra steps:

- Write a gRFC (can be short, but need to tell a complete story) to propose the API change;
- Wait for 14 days for comments and approvals;
- Update the docstring to state under what condition will this method return `None`;
- Update the AIO implementation to return `None`;
- Add tests to guard this behavior.